lib/gems/pending/util/miq-process.rb

```diff
@@ -237,17 +229,23 @@ def self.str_time_to_sec(time_str)
     (t[0].to_i * 3600) + (t[1].to_i * 60) + t[2].to_i
   end
 
+  # Suspend the process +pid+ using a SIGSTOP.
+  #
   def self.suspend_process(pid)
     case Sys::Platform::OS
-    when :windows then Process.process_thread_list[pid].each { |tid| Process.suspend_resume_thread(tid, false) }
+    when :unix
+      Process.kill('STOP', pid)
     else
       raise "Method MiqProcess.suspend_process not implemented on this platform [#{Sys::Platform::IMPL}]"
     end
   end
 
+  # Resume the process +pid+ using a SIGCONT.
+  #
   def self.resume_process(pid)
     case Sys::Platform::OS
-    when :windows then Process.process_thread_list[pid].each { |tid| Process.suspend_resume_thread(tid, true) }
+    when :unix
+      Process.kill('CONT', pid)
```

Comment:
Yeah, this seems much simpler.

However, does there maybe need to be a rescue here as a result?  I think with the `.each`, it would only find the pids if they existed, and no-op'd if they didn't.  That might be what was desired and wanted here.

Reply:
Ok, updated.

Reply:
@NickLaMuro removed, since it's now in more_core_extensions.